### ISSUE TITLE
mDNS service responses are unreliable unless toggling `WiFi.setSleep(false)`

### ISSUE THREAD

**Issue by thorrak** [espressif/arduino-esp32#7156]
### Board

ESP32 (LoLin D32)

### Device Description

LoLin D32 ESP32 module. Additionally tested with a nonstandard ESP32 module with an attached OLED. 

### Hardware Configuration

Nothing else connected

### Version

latest master (checkout manually)

### IDE Name

PlatformIO

### Operating System

N/A (Compiling on MacOS, but issue is unrelated)

### Flash frequency

40mhz

### PSRAM enabled

no

### Upload speed

115200

### Description

I'm attempting to announce a service available on an ESP32 using `MDNS.addService()`, but the ESP32 seems to unreliably respond to queries by service browsers on other devices unless `WiFi.setSleep` is set to `false`. The problem with this workaround is that WiFi/Bluetooth coexistence does not function unless `WiFi.setSleep` is set to `true`.

For the avoidance of doubt, trying to resolve `testhost.local` in the test sketch below seems to work, but searching for `_testservice._tcp.` does not. 

### Sketch

```cpp
#include <Arduino.h>
#include <WiFi.h>
#include <ESPmDNS.h>

const char *ssid = WIFI_SSID;      // Must be set
const char *password = WIFI_PASS;  // Must be set

void setup()
{
  WiFi.disconnect();
  WiFi.mode(WIFI_STA);
  WiFi.begin(ssid, password);
  while (WiFi.status() != WL_CONNECTED)
      delay(1000);

  // This next line is the one that seems to be required for mDNS to work, but means
  // that WiFi and Bluetooth coexistence are not possible 
  WiFi.setSleep(false);

  MDNS.begin("testhost");
  MDNS.addService("_testservice", "_tcp", 80);
}

void loop() {delay(200);}
```

### Debug Message

```plain
N/A
```

### Other Steps to Reproduce

In order to see the mDNS services available on my network, I'm testing using both the [iOS Discovery app](https://apps.apple.com/us/app/discovery-dns-sd-browser/id305441017) and the equivalent [Service Browser](https://play.google.com/store/apps/details?id=com.druk.servicebrowser&hl=en_US&gl=US) app for Android. In both cases, the ESP32 seems to reliably respond to queries when `WiFi.setSleep(false)` is called, and generally miss/ignore queries when it is not. 

To make things easier, I've created a repo here that contains the MVE above, ready to compile in PlatformIO: https://github.com/thorrak/mdns_issue_mve

### I have checked existing issues, online documentation and the Troubleshooting Guide

- [X] I confirm I have checked existing issues, online documentation and Troubleshooting guide.

**Comment by thorrak** [comment]
For what it's worth, the reason I thought to try setting `WiFi.setSleep(false)` was due to comments in [this issue](https://github.com/espressif/esp-idf/issues/9099) on esp-idf by @david-cermak . Not sure if the work @xueyunfei998 or @MaxwellAlan did associated with that issue is related or will ultimately resolve this issue as well.

**Comment by thorrak** [comment]
This appears to be fixed in 2.0.7.
